### PR TITLE
Add AES128 and AES256 XTS KAT tests in POST.

### PR DESCRIPTION
**Pull request by jallisonciq**
KAT vectors copied from GNUTLS.

### CODE REVIEW

**Review by ciq-sahlberg — APPROVED** [review]
LGTM

File: cipher/rijndael.c [hunk 1]
```diff
@@ -1769,6 +1769,146 @@ selftest (void)
   return r;
 }
 
+/* XTS vectors copied from GNUTLS. */
```
Comment (jason-rodri) [inline cipher/rijndael.c:1772]: Do we need to worry about the GnuTLS license?

File: cipher/rijndael.c [hunk 2]
```diff
@@ -1944,6 +2084,11 @@ selftest_fips_128 (int extended, selftest_report_func_t report)
 
   if (extended)
```
Comment (jason-rodri) [inline cipher/rijndael.c:2085]: Do we know if the extended is being called?

**Review by jason-rodri — COMMENTED** [review]
Since we added tests to the extended section make sure they are being executed

File: cipher/rijndael.c (hunk 1 above)
Comment (jallisonciq) [reply cipher/rijndael.c:1772]: The only thing copied is test vectors, I don't believe they can be covered under copyleft licensing.

File: cipher/rijndael.c (hunk 2 above)
Comment (jallisonciq) [reply cipher/rijndael.c:2085]: Yes, extended is set to 1 when called from the global constructor:

```
static void GCC_ATTR_CONSTRUCTOR
_gcry_global_constructor (void)
{
  force_fips_mode = _gcry_fips_to_activate ();
  if (force_fips_mode)
    {
      no_secure_memory = 1;
      global_init ();
      _gcry_fips_run_selftests (1);
```

Comment (jallisonciq) [reply cipher/rijndael.c:2085]: The "1" here: _gcry_fips_run_selftests (1); is 'extended'.

**Comment by jallisonciq** [comment]
> Since we added tests to the extended section make sure they are being executed

Yes, I caught this in gdb and it's being run.

**Review by jason-rodri — APPROVED** [review]

**Pull request merged**